src/unixdgram.c

```diff
@@ -313,10 +313,9 @@ static int meth_getsockname(lua_State *L)
 /*-------------------------------------------------------------------------*\
 * Turns a master unixdgram object into a client object.
 \*-------------------------------------------------------------------------*/
-static const char *unixdgram_tryconnect(p_unix un, const char *path)
+static const char *unixdgram_tryconnect(p_unix un, const char *path, size_t len)
 {
     struct sockaddr_un remote;
-    size_t len = strlen(path);
     if (len >= sizeof(remote.sun_path)) return "path too long";
     memset(&remote, 0, sizeof(remote));
     strcpy(remote.sun_path, path);
```

Comment:
`strcpy()` and NUL-prefixed strings don't work, needs to be changed to something that takes a length. `strncpy()`? `memcpy()`?